server_addon/maya/server/settings/publishers.py

```diff
@@ -309,6 +385,143 @@ class ExtractAlembicModel(BaseSettingsModel):
+            "to the Alembic file.\n"
+
+            "UVs Only - If this flag is present, only uv data for PolyMesh and"
+            " SubD shapes will be written to the Alembic file.\n"
+
+            "Verbose - Prints the current frame that is being evaluated.\n"
+
+            "Whole Frame Geo - Data for geometry will only be written out on "
+            "whole frames.\n"
+
+            "World Space - Any root nodes will be stored in world space.\n"
+
+            "Write Color Sets - Write vertex colors with the geometry.\n"
+
+            "Write Face Sets - Write face sets with the geometry.\n"
+
+            "Write Normals - Write normals with the deforming geometry.\n"
+
+            "Write UV Sets - Write all uv sets on MFnMeshes as vector 2 "
+            "indexed geometry parameters with face varying scope.\n"
+
+            "Write Visibility - Visibility state will be stored in the Alembic"
+            " file. Otherwise everything written out is treated as visible."
+        )
+    )
+    attr: str = SettingsField(
+        title="Custom Attributes",
+        placeholder="attr1;attr2",
+        description=(
+            "Attributes matching by name will be included in the Alembic "
+            "export. Attributes should be separated by semi-colon `;`"
+        )
+    )
+    attrPrefix: str = SettingsField(
+        title="Custom Attributes Prefix",
+        placeholder="prefix1;prefix2",
+        description=(
+            "Attributes starting with these prefixes will be included in the "
+            "Alembic export. Attributes should be separated by semi-colon `;`"
+        )
+    )
+    dataFormat: str = SettingsField(
+        enum_resolver=extract_alembic_data_format_enum,
+        title="Data Format",
+        description="The data format to use to write the file."
+    )
+    melPerFrameCallback: str = SettingsField(
+        title="Mel Per Frame Callback",
+        description=(
+            "When each frame (and the static frame) is evaluated the string "
+            "specified is evaluated as a Mel command."
+        )
+    )
+    melPostFrameCallback: str = SettingsField(
+        title="Mel Post Frame Callback",
+        description=(
+            "When the translation has finished the string specified is "
+            "evaluated as a Mel command."
+        )
+    )
+    pythonPerFrameCallback: str = SettingsField(
+        title="Python Per Frame Callback",
+        description=(
+            "When each frame (and the static frame) is evaluated the string "
+            "specified is evaluated as a python command."
+        )
+    )
+    pythonPostJobCallback: str = SettingsField(
+        title="Python Post Job Callback",
+        description=(
+            "When the translation has finished the string specified is "
+            "evaluated as a python command."
+        )
+    )
+    preRollStartFrame: int = SettingsField(
+        title="Pre Roll Start Frame",
+        description=(
```

Comment:
It may be good in this description to add that `Pre Roll` may need to be added to the Export Flags? (It's a bit unclear at first sight what 'activates' or deactivates this pre roll setting?